### PR TITLE
Add missing e2e tests to back to dashboard navigation

### LINKED ISSUE #31289: [Safety Net] Hold on to query results when you go back-and-forth between dashboards and questions
Epic: https://github.com/metabase/metabase/issues/29500

- [ ] there are no e2e tests for several card types, including action cards
- [ ] there are also no e2e tests for cases when you navigate from and back to a dashboard with loading cards, cards with permissions errors

### PR DESCRIPTION
**Pull request by ranquild**
Closes https://github.com/metabase/metabase/issues/31289

Adds tests for different card types, navigating with cards in the loading state, permission errors.

### CODE REVIEW

File: e2e/test/scenarios/dashboard/dashboard-drill.cy.spec.js [hunk 1]
```diff
@@ -982,156 +966,6 @@ describe("scenarios > dashboard > dashboard drill", () => {
       });
     }
   });
-
```
Comment (ranquild) [inline e2e/test/scenarios/dashboard/dashboard-drill.cy.spec.js:985]: Moved tests to a separate file

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+    cy.findByLabelText(backButtonLabel).should("not.exist");
+  });
+
+  it("should display a back to the dashboard button in table x-ray dashboards", () => {
+    const cardTitle = "Sales per state";
+    cy.visit(`/auto/dashboard/table/${ORDERS_ID}`);
+    cy.wait("@dataset");
+
+    getDashboardCards()
+      .filter(`:contains("${cardTitle}")`)
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:79]: What is the purpose of those `.filter(":contains...` throughout this PR?
Could the same be accomplished with `findByText` (perhaps with `exact: false`)?

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+              cards: [
+                { id: -1, card_id: question_id, ...questionDashcardDetails },
+                { id: -2, ...textDashcardDetails },
+                { id: -3, ...actionDashcardDetails, action_id },
+              ],
+            });
+          },
+        );
+      });
+    });
```
[the remark this answers is not in the document]
Comment (ranquild) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:336]: In cypress you cannot use async/await

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js (hunk 2 above)
Comment (ranquild) [reply e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:79]: No, the goal is to find a card with this title. AFAIK there are no other built-in methods for that.

**Review by oisincoveney — APPROVED** [review]

**Pull request merged**